PyHT6022/HTSDKScope.py

```diff
@@ -161,7 +161,7 @@ def convert_read_data(input_data, scale, scale_points=32.0):
         point_div = scale / scale_points
         out = [0.0 for _ in input_data]
         input_data = [j.value for j in input_data]
-        for j in xrange(0, len(input_data)):
+        for j in range(0, len(input_data)):
```

Comment:
So this is one of the gross corner cases surrounding Python 2 -> Python 3 compatibility that is somewhat irritating. In Python 2, range builds up the whole thing in memory, and is kind of unacceptably slow. Really the ugly (but truly compatible way to solve this is to define at the top):

    try:
        xrange
    except NameError:
        xrange = range

So that both versions have the iter-based range that is appropriate for the for comprehension above.